recipe/recipe.yaml

```
@@ -0,0 +1,48 @@
+context:
+  name: pngquant
+  version: 3.0.3
+
+package:
+  name: ${{ name|lower }}
+  version: ${{ version }}
+
+source:
+  # We cant use a tarball, as we need to clone the submodules
+  git: https://github.com/kornelski/${{ name }}.git
+  tag: ${{ version }}
+
+build:
+  number: 0
+  skip: win # https://github.com/conda-forge/conda-smithy/issues/2239#issuecomment-2898757737
+  script:
+    - if: win
+      then:
+        - cargo install --path . --root "%LIBRARY_PREFIX%"
+      else:
+        - cargo install --path . --root "$PREFIX" --features=static
```

Can you link against `libpng` please? The `build.rs` of `libpng-sys` suggests that this should work (not that complicated).
```suggestion
        - cargo install --path . --root "$PREFIX"
```

---

If i don't manually add `zlib` as a host dependency, pkg-config fails with

```
 │ │ Adding virtual 'pkg-config' package to list of known packages
 │ │ Looking for package 'libpng'
 │ │ Looking for package 'libpng-uninstalled'
 │ │ Reading 'libpng' from file '$PREFIX/lib/pkgconfig/libpng.pc'
 │ │ Parsing package file '$PREFIX/lib/pkgconfig/libpng.pc'
 │ │   line>prefix=$PREFIX
 │ │  Variable declaration, 'prefix' overridden with '$PREFIX'
 │ │   line>exec_prefix=${prefix}
 │ │  Variable declaration, 'exec_prefix' has value '$PREFIX'
 │ │   line>libdir=${exec_prefix}/lib
 │ │  Variable declaration, 'libdir' has value '$PREFIX/lib'
 │ │   line>includedir=${prefix}/include/libpng16
 │ │  Variable declaration, 'includedir' has value '$PREFIX/include/libpng16'
 │ │   line>
 │ │   line>Name: libpng
 │ │   line>Description: Loads and saves PNG files
 │ │   line>Version: 1.6.47
```

This caused the build to silently fall back to the vendored version of libpng, which is what threw me off. Why does pkg-config fail without zlib as an explicit host dependency, even though it's a host dependency of libpng?

---

ping @xhochy

---

Probably because the pkg-config metadata of zlib is only in the `zlib` and not in the `libzlib` package.

---

But since `libpng` has `zlib` as a host dependency, where does `libzlib` come in?

---

Here: https://github.com/conda-forge/zlib-feedstock/blob/3240862e6d0edaa8ec1e624301628c8056950ba0/recipe/meta.yaml#L74 The `run_exports` is `libzlib` as this is the subset of `zlib` you need at runtime.